core/store/src/trie/update.rs

```diff
@@ -200,32 +191,17 @@ impl<'a> Iterator for MergeIter<'a> {
 
 pub struct TrieUpdateIterator<'a> {
     prefix: Vec<u8>,
-    end_offset: Option<Vec<u8>>,
     trie_iter: Peekable<TrieIterator<'a>>,
     overlay_iter: Peekable<MergeIter<'a>>,
 }
 
 impl<'a> TrieUpdateIterator<'a> {
     #![allow(clippy::new_ret_no_self)]
-    pub fn new(
-        state_update: &'a TrieUpdate,
-        prefix: &[u8],
-        start: &[u8],
-        end: Option<&[u8]>,
-    ) -> Result<Self, StorageError> {
+    pub fn new(state_update: &'a TrieUpdate, prefix: &[u8]) -> Result<Self, StorageError> {
         let mut trie_iter = state_update.trie.iter()?;
-        let mut start_offset = prefix.to_vec();
-        start_offset.extend_from_slice(start);
-        let end_offset = match end {
-            Some(end) => {
-                let mut p = prefix.to_vec();
-                p.extend_from_slice(end);
-                Some(p)
-            }
-            None => None,
-        };
-        trie_iter.seek_prefix(&start_offset)?;
-        let committed_iter = state_update.committed.range(start_offset.clone()..).map(
+        trie_iter.seek_prefix(prefix)?;
+        let prefix = prefix.to_vec();
+        let committed_iter = state_update.committed.range(prefix.clone()..).map(
```

Comment:
Not entirely sure what's going on here, but, after just throwing random stuff at the wall, the following seems to compile, while avoding a copy:

```rust
.range::<[u8], _>((Bound::Included(prefix.as_slice()), Bound::Unbounded))
```